x/valset-pref/README.md

```diff
@@ -25,15 +25,41 @@ How does this module work?
 - If the delegator has not set a validator-set preference list then the validator set, then it defaults to their current validator set.
 - If a user has no preference list and has not staked, then these messages / queries return errors.
 
-## Calculations
+## Logic
 
-Staking Calculation 
+### Delegate to validator set
 
-- The user provides an amount to delegate and our `MsgDelegateToValidatorSet` divides the amount based on validator weight distribution.
-  For example: Stake 100osmo with validator-set {ValA -> 0.5, ValB -> 0.3, ValC -> 0.2}
-  our delegate logic will attempt to delegate (100 * 0.5) 50osmo for ValA , (100 * 0.3) 30osmo from ValB and (100 * 0.2) 20osmo from ValC.
+This is pretty straight-forward, theres not really any edge cases here.
 
-UnStaking Calculation 
+The user provides an amount to delegate and our `MsgDelegateToValidatorSet` divides the amount based on validator weight distribution.
+  
+For example: Stake 100osmo with validator-set {ValA -> 0.5, ValB -> 0.3, ValC -> 0.2}
+our delegate logic will attempt to delegate (100 * 0.5) 50osmo for ValA , (100 * 0.3) 30osmo from ValB and (100 * 0.2) 20osmo from ValC.
+
+### Undelegate from validator set 
+
+We can imagine describing undelegate from validator set in two cases:
+- Users existing delegation distribution matches their validator-set preference distribution.
+- Users existing delegation distribution does not match their validator-set preference distribution.
+
+Algorithm for undelegation; 
+// unbond as true to valset ratios as possible. Undelegation should be possible.
```

Comment:
We should add my intuition description for the algorithm back!

Reply:
re-added: https://github.com/osmosis-labs/osmosis/pull/5967/commits/90e9fb2fd747e95310c0b8c26bc518af76fe366e#diff-a6daa0f91df9925cade0547638fc212684689d2b2b8c8f9fa89c1c5a7e505e96R47

Reply:
Is there a reason all of this is commented?

Reply:
@czarcas7ic this was just the pseudocode. I wrote the algorithm here: https://github.com/osmosis-labs/osmosis/pull/5967/files#diff-4422f5e761e99a0e29d2ba7017ece4aa6b486b5ee24b18fb58fcbc9c26a8e7a6R133

following the steps